hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/JetClassLoader.java

```diff
@@ -79,7 +88,75 @@ private InputStream resourceStream(String name) {
         return new InflaterInputStream(new ByteArrayInputStream(classData));
     }
 
+    private URL createResourceURL(String name) {
+        try {
+            return new URL("jet-job-resource", null, -1, name, new JobResourceURLStreamHandler(this));
+        } catch (MalformedURLException e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
     private static boolean isEmpty(String className) {
         return className == null || className.isEmpty();
     }
+
+    private static final class JobResourceURLStreamHandler extends URLStreamHandler {
+
+        private final JetClassLoader jetClassLoader;
+
+        private JobResourceURLStreamHandler(JetClassLoader jetClassLoader) {
+            this.jetClassLoader = jetClassLoader;
+        }
+
+        @Override
+        protected URLConnection openConnection(URL url) throws IOException {
+            return new JobResourceURLConnection(url, jetClassLoader);
+        }
+    }
+
+    private static final class JobResourceURLConnection extends URLConnection {
```

Comment:
this can also be made into a non-static class to avoid passing the instance parameter